### PR TITLE
Refactor: Convert methods with unused return value to void

### PR DESCRIPTION
**Pull request by DanVanAtta**
## Overview
- if a method return value is not used we can convert to void. 
- first commit was done by auto-cleanup, second fixed up the builder code not being used and a void method not making sense.

## Functional Changes
none, cleanup/refactor

### CODE REVIEW

**Comment by ssoloff** [comment]
```
> Task :game-core:spotlessJava FAILED
```

Interesting.  I guess Spotless automatically adds itself to the `check` task, because I didn't do that. :smile:

File: game-core/src/main/java/games/strategy/engine/framework/map/download/MapDownloadController.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public MapDownloadController() {}
    * Return true if all locally downloaded maps are latest versions, false if any can are out of date or their version
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/engine/framework/map/download/MapDownloadController.java]: Javadocs should be updated to reflect removal of return value.

File: game-core/src/main/java/games/strategy/engine/framework/ui/GameChooserEntry.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
   /**
    * Returns a {@link GameData} instance resulting from fully parsing the XML associated with this game.
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/engine/framework/ui/GameChooserEntry.java]: Javadocs should be updated to reflect removal of return value.

File: game-core/src/main/java/games/strategy/sound/ClipPlayer.java [hunk 3]
```diff
@@ -124,13 +124,12 @@ static synchronized ClipPlayer getInstance() {
     return clipPlayer;
   }
 
-  public static synchronized ClipPlayer getInstance(final ResourceLoader resourceLoader) {
+  public static synchronized void getInstance(final ResourceLoader resourceLoader) {
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/sound/ClipPlayer.java]: Possibly outside the scope of this PR, but the method name no longer makes sense.  Would suggest changing it to something like `initialize()` or anything that conveys the notion of creating the `ClipPlayer` singleton if necessary.

File: game-core/src/main/java/games/strategy/ui/SwingComponents.java [hunk 4]
```diff
@@ -147,12 +147,11 @@ public static JTabbedPane newJTabbedPane(final int width, final int height) {
     return tabbedPane;
   }
 
-  public static ButtonGroup newButtonGroup(final JRadioButton... radioButtons) {
+  public static void newButtonGroup(final JRadioButton... radioButtons) {
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/ui/SwingComponents.java]: Also possibly outside the scope of this PR, but, again, this reads like a factory method but doesn't return anything.  Some ideas: `addToNewButtonGroup()`, `groupButtons()`.

Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/ui/SwingComponents.java]: `assignToButtonGroup` sound good?   (528f4f3b5)

File: game-core/src/main/java/games/strategy/sound/ClipPlayer.java (hunk 3 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/sound/ClipPlayer.java]: Good suggestion, though will leave this one out of scope.

File: game-core/src/main/java/games/strategy/ui/SwingComponents.java (hunk 4 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/ui/SwingComponents.java]: This one is in-scope, method and return value should match, good consideration @ssoloff , thank you.

File: game-core/src/main/java/games/strategy/engine/auto/update/TutorialMapCheck.java [hunk 5]
```diff
@@ -8,8 +8,8 @@
 class TutorialMapCheck {
   static void checkForTutorialMap() {
     final MapDownloadController mapDownloadController = ClientContext.mapDownloadController();
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/engine/auto/update/TutorialMapCheck.java]: `mapDownloadController` is now unused (and thus also the `ClientContext` import).

Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/auto/update/TutorialMapCheck.java]: Fixed in: 8a8845793

**Pull request merged**